ruby/command-t/ext/command-t/scanner.c

```diff
@@ -0,0 +1,181 @@
+// Copyright 2010-present Greg Hurrell. All rights reserved.
+// Licensed under the terms of the BSD 2-clause license.
+
+#include <ruby.h>
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>  /* for qsort() */
+#include <string.h>  /* for strncmp() */
+#include <sys/mman.h>
+#include "scanner.h"
+#include "match.h"
+#include "matcher.h"
+#include "heap.h"
+#include "ext.h"
+#include "ruby_compat.h"
+
+// order matters; we want this to be evaluated only after ruby.h
+#ifdef HAVE_PTHREAD_H
+#include <pthread.h> /* for pthread_create, pthread_join etc */
+#endif
```

Comment:
Seems like this is copy-pasta? No use of `pthread_*`, `qsort`, `strncmp` in this file. Haven't checked all the headers included here, but I bet many of these `include` statements can be deleted (`heap.h`, `matcher.h` etc).